recipe_scrapers/_abstract.py

```diff
@@ -14,7 +15,7 @@
 
 # Some sites close their content for 'bots', so user-agent must be supplied
 HEADERS = {
-    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:123.0) Gecko/20100101 Firefox/123.0"
+    "User-Agent": f"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:123.0) Gecko/20100101 recipe-scrapers/{__version__} Firefox/123.0"
```

Comment:
An alternative, perhaps more honest -- but perhaps more-widely-blocked -- user-agent could be something like:

```suggestion
    "User-Agent": f"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:123.0) recipe-scrapers/{__version__}"
```

Basically: declaring `Mozilla/5.0` as many user-agents do, for compatibility reasons, but then only mentioning this library as a component.

I'd like to spend some time to evaluate a random sample of our supported recipe websites -- and perhaps some unsupported recipe sites too -- to get a sense for whether that would be feasible.

For transparency: I have a sense that the [codebase derived from BrowserScope's user-agent parser](https://github.com/ua-parser/uap-core) is one of the most popular Open Source user-agent parsing libraries, and have recently suggested a change in https://github.com/ua-parser/uap-core/pull/598 there to add RecipeRadar (my company, and the primary reason I work on `recipe-scrapers` development) to the identification methods there.  I don't know whether that will be acceptable, but if we do make an update to the `recipe-scrapers` library then I think a corresponding pull request in `uap-core` would also make sense.  I'll get to that eventually, but am mentioning it now so that it's not forgotten about.

Reply:
From testing locally: this value seems to work reasonably well; I've successfully scraped a sample of ~5 recipe URLs where scraper support was recently requested, and one from the README:

  * https://www.bakels.com.au/recipes/donut-glaze/
  * https://www.kitchensanctuary.com/honey-garlic-chicken-skewers/
  * https://www.mundodereceitasbimby.com.pt/pratos-principais-carne-receitas/empadao-de-carne/eyfdj0p4-6f492-762035-cfcd2-j1gkapa1
  * https://kitchenaid.com.au/blogs/kitchenthusiast/hot-cross-buns
  * https://www.allrecipes.com/recipe/158968/spinach-and-feta-turkey-burgers/

That wouldn't be the end of the story if we merge this; I think it's possible that increased traffic as folks upgrade to the library would increase the visibility of the user-agent to the point where it might begin to get blocked.

But ethically, I think it's better to declare what the library (user agent) is, and arguably mixing that with a mention of Firefox is misleading.

That reasoning somewhat locks me out of suggesting that we revert this change at a later date, though, and that could be frustrating.

I'll commit the suggestion to this pull request for now but I think it's worth continuing to consider the implications for a little while longer.